de.dlr.sc.virsat.model.extension.funcelectrical.ui/src/de/dlr/sc/virsat/model/extension/funcelectrical/ui/diagram/features/InterfaceDiagramFeatureProvider.java

```diff
@@ -230,18 +234,22 @@ public IMoveShapeFeature getMoveShapeFeature(IMoveShapeContext context) {
 	
 	@Override
 	public ICustomFeature[] getCustomFeatures(ICustomContext context) {
-	    
 	    PictogramElement[] pictogramElements = context.getPictogramElements();
 		Object object = getBusinessObjectForPictogramElement(pictogramElements[0]);
 		
-		if (object instanceof ABeanStructuralElementInstance) {
-			return new ICustomFeature[] {  new VirSatChangeColorFeature(this)};
-		}
+		List<ICustomFeature> customFeatures = new ArrayList<>();
 		
 		if (object instanceof InterfaceEnd) {
-			return new ICustomFeature[] { new VirsatCategoryAssignmentOpenEditorFeature(this), new InterfaceEndChangeColorFeature(this)};
+			customFeatures.add(new InterfaceEndChangeColorFeature(this));
+		} else {
+			customFeatures.add(new VirSatChangeColorFeature(this));
 		}
-		return super.getCustomFeatures(context);
+		
+		if (object instanceof IBeanCategoryAssignment) {
+			customFeatures.add(new VirsatCategoryAssignmentOpenEditorFeature(this));
+		}
+		
+		return customFeatures.toArray(new ICustomFeature[0]);

```

Comment:
In the previous code we returned super features as a fallback, but now we don't. Does it make any difference?

Reply:
In didnt have any effect right now, but I think it could lead to issues down the road, if somebody were to add a custom feature in the base virsat feature provider.
I changed it to also add all custom features from the parent class.